staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go

```diff
@@ -156,7 +156,12 @@ func (s *store) Get(ctx context.Context, key string, opts storage.GetOptions, ou
 		return storage.NewInternalError(err.Error())
 	}
 
-	return decode(s.codec, s.versioner, data, out, kv.ModRevision)
+	err = decode(s.codec, s.versioner, data, out, kv.ModRevision)
+	if err != nil {
+		metrics.RecordDecodeError(s.groupResourceString)
+		klog.V(4).Infof("Decoding %s \"%s\" failed: %v", s.groupResourceString, key, err)
```

Comment:
can any of the content we're trying to decode leak into `err`? if so, I suspect we shouldn't be logging it (or we could log content of Secret objects, etc)

Reply:
I'm not quite sure if the decode error would leak to `err`, but we've always been logging the err when we called `decode()`. like 
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go#L220-L224
Since it's been there for a long time, I think the leak issue should not be a major concern. WDYT?

Reply:
I still think we need to know what we're logging... can you bit flip a json object and a proto object and see what comes back in the resulting error?

Reply:
Yes I did some reproducing, and the logging looks like this
```
I1208 07:56:17.397972 10 store.go:181] Decoding pods "/pods/default/nginx" failed: illegal base64 data at input byte 3
```